icons/panel-left-close.tsx

```diff
@@ -0,0 +1,101 @@
+'use client';
+
+import type { Transition, Variants } from 'motion/react';
+import { motion, useAnimation } from 'motion/react';
+import type { HTMLAttributes } from 'react';
+import { forwardRef, useCallback, useImperativeHandle, useRef } from 'react';
+import { cn } from '@/lib/utils';
+
+export interface PanelLeftCloseIconHandle {
+  startAnimation: () => void;
+  stopAnimation: () => void;
+}
+
+interface PanelLeftCloseIconProps extends HTMLAttributes<HTMLDivElement> {
+  size?: number;
+}
+
+const defaultTransition: Transition = {
+  times: [0, 0.4, 1],
+  duration: 0.5,
+};
+
+const pathVariants: Variants = {
+  normal: { x: 0 },
+  animate: { x: [0, -1.5, 0] },
+};
+
+const PanelLeftCloseIcon = forwardRef<
+  PanelLeftCloseIconHandle,
+  PanelLeftCloseIconProps
+>(({ onMouseEnter, onMouseLeave, className, size = 28, ...props }, ref) => {
+  const controls = useAnimation();
+  const isControlledRef = useRef(false);
+
+  useImperativeHandle(ref, () => {
+    isControlledRef.current = true;
+    return {
+      startAnimation: () => controls.start('animate'),
+      stopAnimation: () => controls.start('normal'),
+    };
+  });
+
+  const handleMouseEnter = useCallback(
+    (e: React.MouseEvent<HTMLDivElement>) => {
+      if (!isControlledRef.current) {
+        controls.start('animate');
+      } else {
+        onMouseEnter?.(e);
+      }
+    },
+    [controls, onMouseEnter]
+  );
+
+  const handleMouseLeave = useCallback(
+    (e: React.MouseEvent<HTMLDivElement>) => {
+      if (!isControlledRef.current) {
+        controls.start('normal');
+      } else {
+        onMouseLeave?.(e);
+      }
+    },
+    [controls, onMouseLeave]
+  );
+
+  return (
+    <div
+      className={cn(
+        `cursor-pointer select-none p-2 hover:bg-accent rounded-md transition-colors duration-200 flex items-center justify-center`,
+        className
+      )}
+      onMouseEnter={handleMouseEnter}
+      onMouseLeave={handleMouseLeave}
+      {...props}
+    >
+      <svg
+        xmlns="http://www.w3.org/2000/svg"
+        width={size}
+        height={size}
+        viewBox="0 0 24 24"
+        fill="none"
+        stroke="currentColor"
+        strokeWidth="2"
+        strokeLinecap="round"
+        strokeLinejoin="round"
+      >
+        <rect width="18" height="18" x="3" y="3" rx="2" />
+        <path d="M9 3v18" />
+        <motion.path
+          transition={defaultTransition}
+          variants={pathVariants}
+          animate={controls}
+          d="m16 15-3-3 3-3"
+        />
+      </svg>
```

Comment:
Mostly based on `square-chevron-left`. Replaced the svg and kept the same motion.

![image](https://github.com/user-attachments/assets/ee6a3315-aed8-42f1-abc1-97719ed2d025)